### PR TITLE
fix: uniformly set height and border on inputs

### LINKED ISSUE #43528: Input 和 InputNumber 底部基线有 1px 偏差
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/ji-ben-shi-yong-antd-5-7-0-forked-z4lg6z?file=/demo.tsx:0-265)

### Steps to reproduce

```
import React from "react";
import "./index.css";
import { Input, InputNumber } from "antd";

const App: React.FC = () => (
  <>
    <Input placeholder="Basic usage" style={{ width: 50 }} />
    <InputNumber placeholder="Basic usage" />
  </>
);

export default App;
```

### What is expected?

底部基线对齐，无偏差

### What is actually happening?

底部基线有 1px 偏差

| Environment | Info |
| --- | --- |
| antd | 5.7.0 |
| React | 18 |
| System | MacOS 11.5.2 |
| Browser | chrome |

### PR DESCRIPTION
**Pull request by bbb169**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/43528

### 💡 Background and solution

uniformly set height and border on inputs

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix Input and InputNumber align issue.          |
| 🇨🇳 Chinese |  修复 Input 和 InputNumber 行内对齐问题。         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at e0722c8</samp>

This pull request enhances the style and consistency of the `InputNumber` and `Input` components. It removes the redundant border from the `InputNumber` wrapper and adjusts the height calculation of the `Input` component.

### <samp>🤖 Generated by Copilot at e0722c8</samp>

* Remove border from input number wrapper to fix overlap with buttons ([link](https://github.com/ant-design/ant-design/pull/43548/files?diff=unified&w=0#diff-63d0bbc4ed8cb1745da77de552d75aa0fd581ef3eb2d25d0108b2d55eb5f1857L63-R63))
* Add border to input number input to match basic input style and improve consistency ([link](https://github.com/ant-design/ant-design/pull/43548/files?diff=unified&w=0#diff-63d0bbc4ed8cb1745da77de552d75aa0fd581ef3eb2d25d0108b2d55eb5f1857L154-R154))
* Adjust height of basic input to account for border width instead of hard-coding 1px ([link](https://github.com/ant-design/ant-design/pull/43548/files?diff=unified&w=0#diff-3a34112b09233455996fb191066f04b1c3fa01878e9400af4ca8a1dcf6c0106aR142))

### CODE REVIEW

File: components/input/style/index.ts [hunk 1]
```diff
@@ -139,6 +139,7 @@ export const genBasicInputStyle = (token: InputToken): CSSObject => ({
   position: 'relative',
   display: 'inline-block',
   width: '100%',
+  height: token.controlHeight - 2 * token.lineWidth,
```
Comment (afc163) [inline components/input/style/index.ts:142]: 建议改 InputNumber 成 Input 这种实现方式，因为 Input 用的比较多，之前 InputNumber 也是这样的。

Comment (bbb169) [reply components/input/style/index.ts]: ok,已改

https://github.com/ant-design/ant-design/pull/43548/files

File: components/input-number/style/index.ts [hunk 2]
```diff
@@ -147,11 +147,10 @@ const genInputNumberStyles: GenerateStyle<InputNumberToken> = (token: InputNumbe
           '&-input': {
             ...resetComponent(token),
             width: '100%',
-            height: controlHeight - 2 * lineWidth,
-            padding: `0 ${inputPaddingHorizontal}px`,
+            padding: `${inputPaddingVertical}px ${inputPaddingHorizontal}px`,
             textAlign: 'start',
             backgroundColor: 'transparent',
-            border: 0,
+            border: `${lineWidth}px ${lineType} ${colorBorder}`,
```
Comment (afc163) [inline components/input-number/style/index.ts:153]: genBasicInputStyle 里没有这部分代码么，感觉比较奇怪？

**Comment by afc163** [comment]
https://preview-43548-ant-design.surge.sh/components/form-cn#components-form-demo-validate-static
https://preview-43548-ant-design.surge.sh/components/input-number#components-input-number-demo-addon

图片

图片

样式有问题。

**Comment by bbb169** [comment]
> https://preview-43548-ant-design.surge.sh/components/form-cn#components-form-demo-validate-static https://preview-43548-ant-design.surge.sh/components/input-number#components-input-number-demo-addon
> 
> 图片 图片
> 样式有问题。

是border设置在input上导致的，已经把它改回去了

**Comment by afc163** [comment]
看下 argos ci，还有不少问题。

图片

**Review by afc163 — APPROVED** [review]

**Pull request merged**